rest_framework/utils/field_mapping.py

```diff
@@ -203,6 +203,9 @@ def get_relation_kwargs(field_name, relation_info):
     if to_many:
         kwargs['many'] = True
 
+    if to_field:
+        kwargs['to_field'] = to_field
```

Comment:
https://github.com/tomchristie/django-rest-framework/pull/2435/files#diff-80f43677c94659fbd60c8687cce68eafR1007 and the line with your comment below it.